src/WebJobs.Script/Host/AggregateFunctionMetadataProvider.cs

```diff
@@ -62,17 +62,21 @@ public async Task<ImmutableArray<FunctionMetadata>> GetFunctionMetadataAsync(IEn
                     // get function metadata from worker, then validate it
                     rawFunctions = await _dispatcher.GetWorkerMetadata();
 
-                    if (!IsNullOrEmpty(rawFunctions))
+                    if (IsDefaultIndexingRequired(rawFunctions))
+                    {
+                        _logger.LogDebug("Fallback to host indexing as worker denied indexing");
+                        functions = await _hostFunctionMetadataProvider.GetFunctionMetadataAsync(workerConfigs, environment, forceRefresh);
+                    }
+                    else if (!IsNullOrEmpty(rawFunctions))
                     {
                         functions = ValidateMetadata(rawFunctions);
-                        // set up invocation buffers and send load requests
-                        await _dispatcher.FinishInitialization(functions);
                     }
+
+                    // set up invocation buffers and send load requests
+                    await _dispatcher.FinishInitialization(functions);
                 }
-                if (!workerIndexing || IsDefaultIndexingRequired(rawFunctions))
+                else
                 {
-                    // If worker denies indexing then falling back to the host for Indexing
-                    _logger.LogDebug("Fallback to host indexing as worker denied indexing");
                     functions = await _hostFunctionMetadataProvider.GetFunctionMetadataAsync(workerConfigs, environment, forceRefresh);
```

Comment:
What scenario would reach here? both if and else call `_hostFunctionMetadataProvider.GetFunctionMetadataAsync`

Reply:
Host fallback scenario will reach to this _if block_. The final _else_ is for a case where workerIndexing is not enabled at all. 

For host fallback, I still need to do `FinishInitialization(functions);` which is not needed for legacy flow.